docs/philosophy.rst

```diff
@@ -180,3 +180,33 @@ With *dwave-optimization* the dot-product example can be expressed as
     b = model.constant([...])
 
     value = (a * b).sum()
+
+.. _optimization_philosophy_tensor_programming_dynamic:
+
+Dynamically Sized Arrays
+------------------------
+
+*dwave-optimization* does differ significantly from NumPy is one significant
+way: it allows arrays to be *dynamic*. Dynamic arrays are arrays with a
+state-dependent size.
+
+For example, consider a set variable
+(as constructed by :meth:`~dwave.optimization.model.Model.set`).
+The domain of the set variable are any subsets of :math:`[0, n)`.
```

Comment:
It's unclear to me that the domain is as relevant here as the fact that the size varies between ``min_size`` and ``max_size``.